ethereum/spec/build.gradle

```diff
@@ -1,6 +1,7 @@
 dependencies {
     implementation project(':ssz')
     implementation project(':ethereum:datastructures')
+    implementation project(':infrastructure:collections')
```

Comment:
Do we need this?  I don't see any related imports.

Reply:
i can remove it and see what broke

Reply:
```
/Users/paulharris/IdeaProjects/artemis/ethereum/spec/src/main/java/tech/pegasys/teku/spec/util/BeaconStateUtil.java:166: error: cannot access Cache
        .getBeaconProposerIndex()
                               ^
  class file for tech.pegasys.teku.infrastructure.collections.cache.Cache not found
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error
```